cmd/dp-import-tracker/main.go

```diff
@@ -0,0 +1,256 @@
+package main
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/ONSdigital/dp-import-tracker/api"
+	"github.com/ONSdigital/dp-import-tracker/schema"
+	"github.com/ONSdigital/go-ns/kafka"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ian-kent/gofigure"
+)
+
+type config struct {
+	NewInstanceTopic          string   `env:"INPUT_FILE_AVAILABLE_TOPIC" flag:"input-file-available-topic" flagDesc:"topic name for import file available events"`
+	ObservationsInsertedTopic string   `env:"IMPORT_OBSERVATIONS_INSERTED_TOPIC" flag:"observations-inserted-topic" flagDesc:"topic name for increment of inserted observations"`
+	Brokers                   []string `env:"KAFKA_ADDR" flag:"kafka-addr" flagDesc:"topic name for import file available events"`
+	ImportAddr                string   `env:"IMPORT_ADDR" flag:"import-addr" flagDesc:"The address of Import API"`
+	ImportAuthToken           string   `env:"IMPORT_AUTH_TOKEN" flag:"import-auth-token" flagDesc:"Authentication token for access to import API"`
+}
+
+type inputFileAvailable struct {
+	FileURL    string `avro:"file_url"`
+	InstanceID string `avro:"instance_id"`
+}
+
+type insertedObservationsEvent struct {
+	InstanceID                   string `json:"instance_id" avro:"instance_id"`
+	NumberOfObservationsInserted int32  `json:"number_of_inserted_observations" avro:"observations_inserted"`
+}
+
+type trackedInstance struct {
+	totalObservations         int64
+	observationsInsertedCount int64
+	jobID                     string
+}
+
+type trackedInstanceList map[string]trackedInstance
+
+var checkForCompleteInstancesTick = time.Millisecond * 2000
+
+// updateInstanceFromImportAPI updates a specific import instance with the current counts of expected/complete observations
+func (trackedInstances trackedInstanceList) updateInstanceFromImportAPI(client *http.Client, importAPIURL string, instanceID string) error {
+	instanceFromAPI, err := api.GetInstance(client, importAPIURL, instanceID)
+	if err != nil {
+		return err
+	}
+
+	tempCopy := trackedInstances[instanceID]
+
+	if obsCount := instanceFromAPI.NumberOfObservations; obsCount > 0 && obsCount != trackedInstances[instanceID].totalObservations {
+		tempCopy.totalObservations = obsCount
+	}
+	if observationsInsertedCount := instanceFromAPI.TotalInsertedObservations; observationsInsertedCount > 0 && observationsInsertedCount != trackedInstances[instanceID].observationsInsertedCount {
+		tempCopy.observationsInsertedCount = observationsInsertedCount
+	}
+	if importID := instanceFromAPI.Job.ID; importID != trackedInstances[instanceID].jobID {
```

Comment:
This logic could be clearer or have a comment to clarify we're checking for empty instance.jobID not overwriting exisiting ids